src/token/token.controller.ts

```diff
@@ -8,29 +8,52 @@ import {
   Res,
   Post,
   Body,
+  UploadedFile,
+  UseInterceptors,
 } from '@nestjs/common';
+import { FileInterceptor } from '@nestjs/platform-express';
 import { TokenService } from './token.service';
 import { TokenDto } from './token.dto';
 import { TokenResultDto } from './tokenResult.dto';
-import { Response } from 'express';
 import { GrantBody } from 'openid-client';
+import { join } from 'path';
+import { Express, Response } from 'express';
+import { createReadStream, promises as fs } from 'fs';
 
 @Controller('token')
 export default class TokenController {
   constructor(private readonly tokenService: TokenService) {}
 
   @Get('decode')
   @Render('decode')
-  async get() {
+  async get(@Query('schema') schema_s: string) {
+    let empty_schemas;
+    if (schema_s === undefined) {
+      empty_schemas = [''];
+    } else {
+      empty_schemas = [schema_s, ''];
+    }
+    const uploaded_schemas = await fs.readdir('schema/token');
+    const schemas = empty_schemas.concat(uploaded_schemas.filter((x) => { return x !== schema_s; }));
     return {
       message: 'Please enter the wanted information!',
       showResults: false,
+      schemas: schemas,
     };
   }
 
   @Post('decode')
   @Render('decode')
   async post(@Body() tokenDto: TokenDto) {
+    let empty_schemas;
+    const schema_s = tokenDto.schema;
+    if (schema_s === undefined) {
+      empty_schemas = [''];
+    } else {
+      empty_schemas = [schema_s, ''];
+    }
+    const uploaded_schemas = await fs.readdir('schema/token');
```

Comment:
If we don't validate the token against a schema, it is not necessary to load all the schema-files. Maybe just move this to inside an if-statement and only call this if necessary?

Reply:
We also need `uploaded_schemas` for displaying the available schemata in the frontend